Add SSH certificates documentation (#4592)
theforeman/foreman-documentation: guides/common/modules/proc_authenticating-hosts-with-ssh-certificates.adoc

:_mod-docs-content-type: PROCEDURE

> [id="authenticating-hosts-with-ssh-certificates_{context}"]

Reviewer: ```suggestion
[id="authenticating-hosts-with-ssh-certificates"]
```

= Authenticating hosts with SSH certificates

You can configure {SmartProxy} to verify host identity using SSH host certificates.
When enabled, {SmartProxy} verifies that hosts present valid certificates signed by a trusted Certificate Authority (CA) during SSH connections.

> You can configure the {SmartProxy} to trust host certificates signed by multiple CAs.

Reviewer 2: ```suggestion
Your {SmartProxy} can use SSH host certificates to verify host identity.
In this configuration, {SmartProxy} verifies that hosts present valid certificates signed by a trusted Certificate Authority (CA) during SSH connections.
Your {SmartProxy} can trust host certificates signed by multiple CAs.
```
This eliminates some unnecessary words ("configure") and hopefully prevents potential vagueness.

> [WARNING]

Reviewer 2: ```suggestion
[IMPORTANT]
```
Warning should IMO be reserved for security gaps. This is probably less severe.

====

> After you enable host certificate verification on {SmartProxy}, all hosts that use this {SmartProxy} for remote execution can no longer authenticate unless they are configured to use host certificates.

Reviewer 2: ```suggestion
After you enable host certificate verification on {SmartProxy}, all hosts that use this {SmartProxy} for remote execution can authenticate only if they are configured to use host certificates.
```

====

.Prerequisites

> * You have a Certificate Authority for signing SSH host certificates.

Reviewer 2: ```suggestion
* You have a CA for signing SSH host certificates.
```

.Procedure

> . On each host that uses this {SmartProxy} for remote execution, perform the following steps:

Reviewer 2: It would be nice to replace the fluffy-ish "the following steps" with a more specific description of the action a user can expect. So perhaps:

```suggestion
. On each host that uses this {SmartProxy} for remote execution, prepare the SSH certificate for the host:
```

.. Export one of the host's public keys located in `/etc/ssh/` to the CA server.
.. Sign the public key with your CA to create an SSH host certificate.
.. Import the SSH certificate back to the host.
The file name must be `_My_Host_Private_SSH_Key_-cert.pub`.

> The location must be the same as the location of the private key.

Author: The file gets created automatically with the correct name, this should be something like a warning to not change it.

> The location must be the same as the location of the private key.

Author: Applied.
